commons/src/main/java/org/infinispan/commons/dataconversion/Encoder.java

```diff
@@ -0,0 +1,30 @@
+package org.infinispan.commons.dataconversion;
+
+/**
+ * Used to convert data between read/write format and storage format.
+ * @since 9.1
+ */
+public interface Encoder {
```

Comment:
@gustavonalle, I was pondering if the ```Encoder``` interface should really make any distinction between key and value. In theory it should not care whether the material it encodes represents a key or a value. My instinct was to have two encoders, one in charge of the key, the other for the value, and ```AdvancedCache.withEncoder``` would take 2 encoder args, and there would also be a short-cut variant taking only one encoder to be used for both key and value.